userspace/libsinsp/events/sinsp_events.cpp

```diff
@@ -151,10 +151,18 @@ libsinsp::events::set<ppm_event_code> libsinsp::events::all_event_set()
 libsinsp::events::set<ppm_sc_code> libsinsp::events::event_set_to_sc_set(const set<ppm_event_code>& events_of_interest)
```

Comment:
@FedeDP there seems to be still a bug in here. Please see the new unit tests I added in https://github.com/falcosecurity/libs/pull/913

```
/* This test asserts that `sc_set_to_event_set <-> event_set_to_sc_set` back and forth mapping is correct
 */
TEST(interesting_syscalls, sc_set_to_event_set__event_set_to_sc_set)
{
	auto a = libsinsp::events::names_to_sc_set({"open", "openat"});
	auto mapping1a = libsinsp::events::sc_set_to_event_set(a);
	auto mapping2a = libsinsp::events::event_set_to_sc_set(mapping1a); // This fails, expected is 2, but we get 4 hence we don't correctly map from _E and _X back to just SC
	ASSERT_EQ(mapping2a.size(), a.size());
	ASSERT_EQ(mapping2a.size(), 2);

	auto b = libsinsp::events::names_to_event_set({"open", "openat"});
	auto mapping1b = libsinsp::events::event_set_to_sc_set(b);
	auto mapping2b = libsinsp::events::sc_set_to_event_set(mapping1b);
	ASSERT_EQ(mapping2b.size(), b.size());
	ASSERT_EQ(mapping2b.size(), 6);
}

```

Reply:
I don't think it is a bug: we are smart enough to understand that the "open", "openat" set needs "sys_enter" and "sys_exit" too.
Imagine a consumer wants to know, given some event names, the set of PPM_SC that she/he must enable during `sinsp::open_kmod()`; without that logic, the resulting set would be uneffective because no `sys_enter` nor `sys_exit` tracepoint would be attached.

I think this is the expected and safest behavior. Perhaps we can add a documentation note?